lib/task.js

```diff
@@ -151,20 +153,133 @@ function buildDeviceBindings(devices, taskScopes) {
   return deviceBindings;
 }
 
+export class Reclaimer {
+  constructor(runtime, task, primaryClaim, claim) {
+    this.runtime = runtime;
+    this.task = task;
+    this.primaryClaim = primaryClaim;
+    this.claim = claim;
+    this.stopped = false;
+
+    // start things off
+    this.scheduleReclaim();
+  }
+
+  /**
+   * Stop reclaiming.  If a reclaim is already in progress, it will complete.
+   */
+  stop() {
+    this.stopped = true;
+    this.clearClaimTimeout();
+  }
+
+  /**
+   * Determine when the right time is to issue another reclaim then schedule it
+   * via setTimeout.
+   */
+  scheduleReclaim() {
+    let claim = this.claim;
+
+    // Figure out when to issue the next claim...
+    let takenFor = (new Date(claim.takenUntil) - new Date());
+    let nextClaim = takenFor / this.runtime.task.reclaimDivisor;
+
+    // This is tricky ensure we have logs...
+    this.log('next claim', {time: nextClaim});
+
+    // Figure out when we need to make the next claim...
+    this.clearClaimTimeout();
+
+    this.claimTimeoutId =
+      setTimeout(function() {
+        async () => {
+          await this.reclaimTask();
+        }()
+      }.bind(this), nextClaim);
+  }
+
+  /**
+   * Clear next reclaim if one is pending...
+   */
+  clearClaimTimeout() {
+    if (this.claimTimeoutId) {
+      clearTimeout(this.claimTimeoutId);
+      this.claimTimeoutId = null;
+    }
+  }
+
+  /**
+   * Reclaim the current task and schedule the next reclaim...
+   */
+  async reclaimTask() {
+    let task = this.claim.task;
+
+    if (this.stopped) {
+      return;
+    }
+
+    this.log('reclaiming task');
+
+    try {
+      this.claim = await this.runtime.queue.reclaimTask(
+                                this.claim.status.taskId, this.claim.runId);
+      // reclaim does not return the task, so carry that forward from the previous
+      // claim
+      this.claim.task = task;
+    } catch (e) {
+      let errorMessage = `Could not reclaim task. ${e.stack || e}`;
+      this.log('error reclaiming task', {error: errorMessage});
+
+      // If this is not the primary claim, just stop trying to reclaim.  The task
+      // will attempt to resolve it as superseded, and fail, but the primary task
+      // and the other superseded tasks will still be resolved correctly.
+      if (this.claim != this.primaryClaim) {
```

Comment:
So here this doesn't evaluate to true when there is only one task.  I believe perhaps it could be because both are objects, are not the same objects in memory, so they do not evaluate to true.